quill-sql/src/test/scala/io/getquill/ported/quotationspec/InfixTest.scala

```diff
@@ -148,14 +148,14 @@ class InfixTest extends Spec with Inside {
         q must matchPattern {
           case Quoted(
             QuotationTag(idA),
-            List(),
+            List(EagerPlanter("foo", _, idB1)),
             List(
               QuotationVase(
-                Quoted(Infix(List("dyn1 || ", ""), List(ScalarTag(idB1, _)), false, false, QV), List(EagerPlanter("foo", _, idB)), Nil),
+                Quoted(Infix(List("dyn1 || ", ""), List(ScalarTag(idB, _)), false, false, QV), List(EagerPlanter("foo", _, idB2)), Nil),
                 idA1
               )
             )
-          ) if (idA == idA1 && idB == idB1) =>
+          ) if (idA == idA1 && idB == idB1 && idB == idB2) =>
```

Comment:
If I understand correctly how `q` is parsed, where `q` is
```scala
        val q = quote {
          sql"#$a || ${lift("foo")}".as[String]
        }
```
then
- The "lift" macro fires first which returns `EagerPlanter("foo", ...).unquote`
- Quote macro is called which passes `'{EagerPlanter("foo").unquote}` to the parser.
- `QuotationParser` fires first on `UprootableUnquote` where expr is
`EagerPlanter("foo")` and adds `EagerPlanter("foo")` to `Lifts.lifts`
- `InfixParser` then fires, as part of parsing `ExtractLifts` is called which _gets_ the
"foo" lift.

which, given that we are now "getting" lifts from the parser, makes sense the `EagerPlanter("foo")` would be now included in the `lifts` of the returned `Quoted`.

Reply:
This logic sounds right. The actual EagerPlanter(foo) is supposed to live inside the QuotationVase. I don’t think the ExtractLifts inside InfixParser is even necessary anymore and you should try to remove it.

Reply:
> I don’t think the ExtractLifts inside InfixParser is even necessary anymore and you should try to remove it.
I'll start looking into this

Reply:
@deusaquilus Indeed, you are right; simply getting the "lifts" and "pluckableUnquotes" from the `Lifts` context after parsing the `Param`s seems to work: all of the tests with `InfixTest` pass. 

Just pushed up a commit with this change.